### PR TITLE
Final niriss im

### PR DESCRIPTION
**Pull request by rizeladiaz**
This notebook checklist has been made available to us by the [the Notebooks For All team](https://github.com/Iota-School/notebooks-for-all/blob/main/resources/event-hackathon/notebook-authoring-checklist.md).
Its purpose is to serve as a guide for both the notebook author and the technical reviewer highlighting critical aspects to consider when striving to develop an accessible and effective notebook.

### The First Cell

- [x] The title of the notebook in a first-level heading (eg. `<h1>` or `# in markdown`).
- [x] A brief description of the notebook.
- [x] A table of contents in an [ordered list](https://www.markdownguide.org/basic-syntax/#ordered-lists) (`1., 2.,` etc. in Markdown).
- [x] The author(s) and affiliation(s) (if relevant).
- [x] The date first published.
- [x] The date last edited (if relevant).
- [ ] A link to the notebook's source(s) (if relevant).

### The Rest of the Cells

- [x] There is only one H1 (`#` in Markdown) used in the notebook.
- [x] The notebook uses other [heading tags](https://www.markdownguide.org/basic-syntax/#headings) in order (meaning it does not skip numbers). 

## Text

- [x] All link text is descriptive. It tells users where they will be taken if they open the link.
- [x] All acronyms are defined at least the first time they are used. 
- [x] Field-specific/specialized terms are used when needed, but not excessively.

## Code

- [x] Code sections are introduced and explained before they appear in the notebook. This can be fulfilled with a heading in a prior Markdown cell, a sentence preceding it, or a code comment in the code section.
- [x] Code has explanatory comments (if relevant). This is most important for long sections of code.
- [x] If the author has control over the syntax highlighting theme in the notebook, that theme has enough color contrast to be legible.
- [x] Code and code explanations focus on one task at a time. Unless comparison is the point of the notebook, only one method for completing the task is described at a time.

### Images

- [ ] All images (jpg, png, svgs) have an [image description](https://www.w3.org/WAI/tutorials/images/decision-tree/). This could be
    - [ ] [Alt text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt) (an `alt` property) 
    - [ ] [Empty alt text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt#decorative_images) for decorative images/images meant to be skipped (an `alt` attribute with no value)
    - [ ] Captions
    - [ ] If no other options will work, the image is decribed in surrounding paragraphs.

- [ ] Any [text present in images](https://www.w3.org/WAI/WCAG21/Understanding/images-of-text.html) exists in a text form outside of the image (this can be alt text, captions, or surrounding text.)

### Visualizations

- [ ] All visualizations have an image description. Review the previous section, Images, for more information on how to add it.
- [ ] [Visualization descriptions](http://diagramcenter.org/specific-guidelines-e.html) include
    - [ ] The type of visualization (like bar chart, scatter plot, etc.)
    - [ ] Title
    - [ ] Axis labels and range
    - [ ] Key or legend
    - [ ] An explanation of the visualization's significance to the notebook (like the trend, an outlier in the data, what the author learned from it, etc.)

- [ ] All visualizations and their parts have [enough color contrast](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) ([color contrast checker](https://webaim.org/resources/contrastchecker/)) to be legible. Remember that transparent colors have lower contrast than their opaque versions.
- [ ] All visualizations [convey information with more visual cues than color coding](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html). Use text labels, patterns, or icons alongside color to achieve this.
- [ ] All visualizations have an additional way for notebook readers to access the information. Linking to the original data, including a table of the data in the same notebook, or sonifying the plot are all options.

### CODE REVIEW

**Comment by rizeladiaz** [comment]
Changes to Notebook:
———————————————————

    Notebook chaged name to match format
    changed the requirement files to set numpy <2.0 and remove CRDS call from there. Changed version of the pipieline to run 1.14.0
    Added author field.
    NIRISS Team reviewed Recent Changes section
    Added section to configure CRDS
    Changed the organization to agree with that of the template notebook. Mainly move the import packages after configuration of CRDS.
    Added demo mode to import from astroquery
    Moved information about restricting data to one filter to the section talking about getting the data from MAST
    Removed sections for getting the data from Box and added sections to get the data from MAST in demo mode
    Added section to set the directories as the MRS notebook
    Removed mention of WFSS and AMI because these do not apply to this notebook.
    Removed section about read-the-docs pipeline documentation as it repeats what is provided in JDox and could become out of sync.
    Added time benchmark counters for all stages. NIRISS changed them to be integer seconds.
    NIRISS team changed the way the reference files are printed to a separate cell
    Added In Detector 1 a new section to define a dictionary with the steps that run and examples of how to override parameters or reference files via a dictionary.
    Added dictionary and parameters for Image 2
    NIRISS team did check the parameters for Image2 run
    Added dictionary and parameters for Image 3
    NIRISS team did check parameters for Image3 run
    Moved to save the association Image3 file in the ObsXXX directory and changed cal files get the full path rather than relative paths (which was crashing the run).
    Moved plotting or data visualization to the end

**Review by sosey — CHANGES_REQUESTED** [review]
see comments inline.
do we want to turn on  a notebook run to see how this manages in actions?

File: notebooks/NIRISS/JWpipeNB-niriss-imaging.ipynb
Comment (rizeladiaz) [reply notebooks/NIRISS/JWpipeNB-niriss-imaging.ipynb:18]: I don't know. It didn't have author and those are the names that Steph put.

File: notebooks/NIRISS/JWpipeNB-niriss-imaging.ipynb
Comment (rizeladiaz) [reply notebooks/NIRISS/JWpipeNB-niriss-imaging.ipynb:20]: @drlaw1558 I will add the footer for the next version

Comment (drlaw1558) [reply notebooks/NIRISS/JWpipeNB-niriss-imaging.ipynb:20]: The main difference between 1.14.0 and 1.14.1 should be in the numpy version, with default numpy for 1.14.0 not being fully compatible.  If it's possible to update more completely to 1.15.1 we may as well.

**Comment by bhilbert4** [comment]
@rizeladiaz while modifying a copy of this notebook to work on NIRCam imaging this afternoon, I found that your table of contents links don't work.

**Comment by rizeladiaz** [comment]
> @rizeladiaz while modifying a copy of this notebook to work on NIRCam imaging this afternoon, I found that your table of contents links don't work.

I just checked and they work, except for the last one. The title got changed for that one and I didn't realize it until now. Thanks for spotting that.  Try to save your notebook and try again the links. These should work. If not check the latest version and see if the format matches what you have.

**Review by sosey — APPROVED** [review]
lgtm

**Comment by drlaw1558** [comment]
Is this ready to merge?  I know @slamassa is aiming to update the pipeline version for the notebook, and that would be cleaner to separate from this initial PR.

**Comment by rizeladiaz** [comment]
Yes, this can be merged. This version already is using 1.15.1. I cannot merge it because it says it is blocked. I think you have to approve?

**Review by drlaw1558 — APPROVED** [review]

File: notebooks/NIRISS/JWpipeNB-niriss-imaging.ipynb
Comment (drlaw1558) [reply notebooks/NIRISS/JWpipeNB-niriss-imaging.ipynb]: I guess this didn't get officially resolved, but I believe we're all set for now unless @sosey sees other issues.

**Comment by drlaw1558** [comment]
I think it may have been blocked by an unresolved conversation- that was just a version discussion though so I resolved it.

**Comment by rizeladiaz** [comment]
I cannot merge it because it still says "Required statuses must pass before merging." Megan, could you help with this? The PEP8 requirements pass, so not sure what else it wants.

**Comment by drlaw1558** [comment]
@sosey Following up on our conversation earlier- it looks like the most recent NIRISS notebook PEP8 check actually passed according to the workflow runs, it's older versions that had problems.  Seems like this status might be getting blocked by any failing PEP8 check in the actions history rather than the most recent commit on this PR itself?

**Comment by drlaw1558** [comment]
Any update on the runners here?  I'm about to send out emails to team leads about the upcoming AAS meeting, and it would be great to have the NIRISS notebook merged as another example case they can look at.

**Comment by sosey** [comment]
We still dont have a manual execute for PRs, but I'll merge this. I'll turn on automated execution for future PR

**Pull request merged**
